### PR TITLE
[MetaSchedule] Enhance Conv2d NCHW Winograd Schedule Rules

### PR DESCRIPTION
**Pull request by zxybazh**
This PR introduces some new trials to expand the search space for Winograd Conv2d Workload by writing block level schedule rules. Tests will also be included.

### CODE REVIEW

**Comment by zxybazh** [comment]
@tvm-bot rerun

File: src/meta_schedule/schedule_rule/winograd.cc [hunk 1]
```diff
@@ -92,6 +124,50 @@ TVM_REGISTER_GLOBAL("meta_schedule.winograd_data_pack.llvm")
...
+      Array<LoopRV> split0 = sch->Split(loops[2], {NullOpt, tile_size.value()});
+      Array<LoopRV> split1 = sch->Split(loops[3], {NullOpt, tile_size.value()});
+      sch->Reorder({split0[0], split1[0], split0[1], split1[1]});
+
+      // compute_at
+      sch->ComputeAt(inverse, /*loop_rv=*/split1[0],
+                     /*preserve_unit_loops=*/true);
+      if (OL.defined()) {
+        while (sch->GetConsumers(OL).size() > 0) {
+          BlockRV next_OL = GetOnlyConsumer(sch, OL);
```
Comment (vinx13) [inline src/meta_schedule/schedule_rule/winograd.cc:153]: Will consumers be handled automatically by auto_inline rule? If so, this and `SetScope` are not necessary

File: src/meta_schedule/schedule_rule/winograd.cc [hunk 2]
```diff
@@ -74,6 +80,32 @@ inline LoopRV ScheduleDataPack(Schedule sch, BlockRV block) {
...
+    if (*i <= 16) {
+      sch->Unroll(loops[1]);
+    }
+  }
+  sch->Unroll(loops[4]);
+  sch->Unroll(loops[5]);
+
+  Array<ExprRV> factors = sch->SamplePerfectTile(loops[3], /*n=*/2, /*max_innermost_factor=*/64);
+  Array<LoopRV> split =
+      sch->Split(loops[3], /*factors=*/{factors[0], factors[1]}, /*preserve_unit_loops=*/true);
```
[the remark this answers is not in the document]
Comment (zxybazh) [reply src/meta_schedule/schedule_rule/winograd.cc:102]: Hi, it seems `factors` are type of `Array<PrimExpr>` while the argument here requires `factors` to be `Array<Optional<PrimExpr>>` so IIUC we may keep it this way.

**Comment by zxybazh** [comment]
Hi Wuwei, I have removed bgemm block schedule and use MultiLevelTiling instead. It reproduced same performance in conv2d test, I’m running e2e tests as well. With other issues also addressed, would you please take another look? Thanks!

**Review by vinx13 — APPROVED** [review]

File: tests/python/unittest/test_meta_schedule_space_cuda.py [hunk 3]
```diff
@@ -1239,6 +1239,90 @@ def cbr_0(data: T.Buffer[(1, 224, 224, 3), "float32"], kernel: T.Buffer[(7, 7, 3
     )
 
 
+def test_cuda_tbg():
```
Comment (vinx13) [inline tests/python/unittest/test_meta_schedule_space_cuda.py:1242]: this test case is not related

File: tests/python/unittest/test_meta_schedule_space_cuda.py [hunk 4]
```diff
@@ -1378,6 +1456,9 @@ def winograd_nchw_conv2d(data: T.Buffer[(1, 64, 224, 224), "float32"], kernel: T
         space_generator=ms.space_generator.PostOrderApply(),
         sch_rules="default",
     ).generate_design_space()
+    sch = actual[0]
+    print(sch.mod.script())
+    print(sch.trace)
```
Comment (vinx13) [inline tests/python/unittest/test_meta_schedule_space_cuda.py:1461]: remove it

File: tests/python/unittest/test_meta_schedule_space_cuda.py (hunk 3 above)
Comment (zxybazh) [reply tests/python/unittest/test_meta_schedule_space_cuda.py:1242]: This test case was moved to the right position before the PR, it got moved after the conv2d winograd testcase because of rebase. You can see there’s no change in the `files changed` tab page.
